sksurv/compare.py

```diff
@@ -75,6 +75,7 @@ def compare_survival(y, group_indicator, return_stats=False):
     observed = numpy.zeros(n_groups, dtype=numpy.int_)
     expected = numpy.zeros(n_groups, dtype=numpy.float_)
     covar = numpy.zeros((n_groups, n_groups), dtype=numpy.float_)
+    group_eye = numpy.eye(n_groups, dtype=bool)
```

Comment:
Since you are only using it for indexing `covar`, using `numpy.diag_indices` makes more sense to me.

Reply:
I didn't know such thing exist, sure, way less memory consumption